plugins/httpapi/mso.py

```diff
@@ -236,33 +236,38 @@ def _response_to_json(self, response_data):
         # JSONDecodeError only available on Python 3.5+
         except Exception as e:
             # Expose RAW output for troubleshooting
-            self.error = dict(code=-1, message="Unable to parse output as JSON, see 'raw' output. {0}".format(e))
+            self.error = dict(code=self.status, message="Unable to parse output as JSON, see 'raw' output. {0}".format(e))
             self.info["raw"] = response_text
             return
 
     def _get_login_domain_id(self, domain_name):
         """Get a domain and return its id"""
         if domain_name is None:
             return None
+
+        method = "GET"
+        path = "/mso/api/v1/auth/login-domains"
+        full_path = self.connection.get_option("host") + path
+
         # TODO: Replace response by -
-        response, data = self.send_request("GET", "auth/login-domains")
+        response, data = self.connection.send(path, None, method=method, headers=self.headers)
 
         if data is not None:
             response_data = self._response_to_json(data)
             domains = response_data.get("domains")
             if domains is not None:
                 for domain in domains:
-                    if domain.get("name") != domain_name:
+                    if domain.get("name") == domain_name:
                         if "id" in domain:
                             return domain.get("id")
                         else:
-                            self.error = dict(code=-1, message="Login domain '{0}' is not a valid domain name.".format(domain))
-                            raise ConnectionError(self._verify_response(None, None, None, None))
-                self.error = dict(code=-1, message="Login domain lookup failed for domain '{0}': {1}".format(domain_name, domain))
-                raise ConnectionError(self._verify_response(None, None, None, None))
+                            self.error = dict(code=self.status, message="Login domain lookup failed for domain '{0}': {1}".format(domain_name, domain))
+                            raise ConnectionError(json.dumps(self._verify_response(response, method, full_path, data)))
+                self.error = dict(code=self.status, message="Login domain '{0}' is not a valid domain name.".format(domain_name))
+                raise ConnectionError(json.dumps(self._verify_response(response, method, full_path, data)))
             else:
-                self.error = dict(code=-1, message="Key 'domains' missing from data")
-                raise ConnectionError(self._verify_response(None, None, None, None))
+                self.error = dict(code=self.status, message="Key 'domains' missing from data")
+                raise ConnectionError(json.dumps(self._verify_response(response, method, full_path, data)))
```

Comment:
would it be an option to have less copy/paste code? for example by setting error_message to the string and moving the error and raise to end of the if? something like example below (have not tested it)

```
error_message = ""
if data is not None:
    response_data = self._response_to_json(data)
    domains = response_data.get("domains")
    if domains is not None:
        for domain in domains:
            if domain.get("name") == domain_name:
                if "id" in domain:
                    return domain.get("id")
                else:
                    error_message = "Login domain lookup failed for domain '{0}': {1}".format(domain_name, domain)
        error_message = "Login domain '{0}' is not a valid domain name.".format(domain_name)
    else:
        error_message = "Key 'domains' missing from data"

if error_message:
    self.error = dict(code=self.status, message=error_message)
    raise ConnectionError(json.dumps(self._verify_response(response, method, full_path, data)))
```

Reply:
It goes both ways. By doing the above we increase the number of lines of code. Let me know if you still want to change it.